app/src/main/java/com/github/se/cyrcle/ui/report/ViewReportsScreen.kt

```diff
@@ -0,0 +1,167 @@
+import androidx.compose.material3.FabPosition
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.github.se.cyrcle.R
+import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.model.report.Report
+import com.github.se.cyrcle.model.report.ReportedObjectType
+import com.github.se.cyrcle.model.report.ReportedObjectViewModel
+import com.github.se.cyrcle.model.review.ReviewViewModel
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.navigation.Screen
+import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
+
+@SuppressLint("StateFlowValueCalledInComposition")
+@Composable
+fun ViewReportsScreen(
+    navigationActions: NavigationActions,
+    reportedObjectViewModel: ReportedObjectViewModel,
+    parkingViewModel: ParkingViewModel,
+    reviewViewModel: ReviewViewModel
+) {
+  val currentObject = reportedObjectViewModel.selectedObject.value
+  val context = LocalContext.current
+  val selType = currentObject?.objectType ?: ReportedObjectType.PARKING
+  val successDeleteText = stringResource(R.string.object_deleted)
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            navigationActions,
+            title =
+                if (selType == ReportedObjectType.PARKING) {
+                  stringResource(R.string.parking_reports)
+                } else {
+                  stringResource(R.string.review_reports)
+                })
+      },
+      floatingActionButton = {
+        FloatingActionButton(
+            onClick = {
+              val uidOfObject = currentObject?.objectUID
+              if (uidOfObject != null) {
+                reportedObjectViewModel.deleteReportedObject(uidOfObject)
+                if (selType == ReportedObjectType.PARKING) {
+                  parkingViewModel.deleteParkingByUid(uidOfObject)
+                } else {
+                  reviewViewModel.deleteReviewById(uidOfObject)
+                }
+                Toast.makeText(context, successDeleteText, Toast.LENGTH_LONG).show()
+                navigationActions.navigateTo(Screen.ADMIN)
+              }
+            },
+            modifier = Modifier.padding(16.dp),
+            containerColor = MaterialTheme.colorScheme.errorContainer) {
+              Text(
+                  text = stringResource(R.string.delete_object),
+                  color = MaterialTheme.colorScheme.onPrimary,
+                  style = MaterialTheme.typography.bodyLarge)
+            }
+      },
+      floatingActionButtonPosition = FabPosition.End) { paddingValues ->
```

Comment:
This can be simplified to avoid the repetition, especially if reports.isEmpty is true : 

`floatingActionButtonPosition = FabPosition.End) { paddingValues ->
    Box(modifier = Modifier.padding(paddingValues)) {
        val reports = when (selType) {
            ReportedObjectType.PARKING -> parkingViewModel.selectedParkingReports.value
            ReportedObjectType.REVIEW -> reviewViewModel.selectedReviewReports.value
        }

        Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
            if (reports.isEmpty()) {
                Text(
                    text = stringResource(R.string.no_reports_available),
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.align(Alignment.CenterHorizontally)
                )
            } else {
                LazyColumn(
                    modifier = Modifier.fillMaxSize(),
                    contentPadding = PaddingValues(vertical = 8.dp)
                ) {
                    items(reports) { report ->
                        val reportType = when (selType) {
                            ReportedObjectType.PARKING -> Report.Parking(report)
                            ReportedObjectType.REVIEW -> Report.Review(report)
                        }
                        ReportCard(reportType)
                    }
                }
            }
        }
    }
}
`

Reply:
Or if this works, even better :

```
floatingActionButtonPosition = FabPosition.End) { paddingValues ->
    Box(modifier = Modifier.padding(paddingValues)) {
        val reportCards: List<Report> = when (selType) {
            ReportedObjectType.PARKING -> 
                parkingViewModel.selectedParkingReports.value.map { Report.Parking(it) }
            ReportedObjectType.REVIEW -> 
                reviewViewModel.selectedReviewReports.value.map { Report.Review(it) }
        }

        Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
            if (reportCards.isEmpty()) {
                Text(
                    text = stringResource(R.string.no_reports_available),
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.align(Alignment.CenterHorizontally)
                )
            } else {
                LazyColumn(
                    modifier = Modifier.fillMaxSize(),
                    contentPadding = PaddingValues(vertical = 8.dp)
                ) {
                    items(reportCards) { reportCard ->
                        ReportCard(reportCard)
                    }
                }
            }
        }
    }
}

```